src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,277 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static class Node<T> {
+        private T item;
+        private Node<T> prev;
+        private Node<T> next;
+
+        private Node(T element) {
+            this.item = element;
+        }
+    }
+
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    public MyLinkedList() {
+        size = 0;
+        head = null;
+        tail = null;
+    }
+
     @Override
     public void add(T value) {
+        Node<T> elementToAdd = new Node<>(value);
+
+        if (size == 0) {
+            head = elementToAdd;
+            tail = elementToAdd;
+        } else {
+            tail.next = elementToAdd;
+            elementToAdd.prev = tail;
+            tail = elementToAdd;
+        }
+
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index " + index + " does not exist");
+        }
+
+        Node<T> elementToAdd = new Node<>(value);
+
+        if (size == 0) {
+            head = elementToAdd;
+            tail = elementToAdd;
+        } else if (size - 1 == index) {
+            Node<T> elPrevTail = tail.prev;
+            elPrevTail.next = elementToAdd;
+            elementToAdd.prev = elPrevTail;
+
+            tail.prev = elementToAdd;
+            elementToAdd.next = tail;
```

Comment:
The condition `size - 1 == index` seems incorrect for adding an element at a specific index. It should be `index == size - 1` to add at the end of the list. Additionally, the logic here is incorrect as it tries to insert before the tail, which is not the intended behavior for adding at the end.